lib/cequel/record/associations.rb

```diff
@@ -108,14 +108,16 @@ def belongs_to(name, options = {})
                  "belongs_to association must be declared before declaring " \
                  "key(s)"
           end
-          
+
           key_options = options.extract!(:partition)
+          foreign_key_parts = [options.fetch(:foreign_key, [])].flatten
 
           self.parent_association =
             BelongsToAssociation.new(self, name.to_sym, options)
 
-          parent_association.association_key_columns.each do |column|
-            key :"#{name}_#{column.name}", column.type, key_options
+          parent_association.association_key_columns.each_with_index do |column, i|
+            foreign_key = foreign_key_parts[i].nil? ? "#{name}_#{column.name}" : foreign_key_parts[i]
+            key foreign_key.to_sym, column.type, key_options
```

Comment:
It is closely related to the comment above. As i read this code it seems that someone could write `... foreign_key: [nil, :parent_id]` and it would come out as `[:foo_account_id, :parent_id]`. That seems a little weird.

It would be better, in my view, to require the `:foreign_key`  to be either entirely absent or fully formed and complete.